### DISCUSSION TITLE
Import error pyart

### DISCUSSION THREAD

**Discussion by rakeshghosh313** [ARM-DOE/pyart#1538] [General]
Whenever I am trying to import my stable pyart library , it is showing error "module 'numpy' has no attribute 'typeDict'"
How to solve the issue??

**Comment by mgrover1** [comment]
which Py-ART version are you using? And which version of numpy?

**Reply by rakeshghosh313** [reply]
Hi..!
Thanks for the question.
Recently, I have updated both libraries. 
I have updated the Py-ART 1.18.0
I am using the numpy 1.24.4
Spyder (python 3.8) is showing this issue.

**Reply by zssherman** [reply]
@rakeshghosh313 Is there error being called from within Py-ART? I'm not seeing a typeDict in Py-ART itself, but in or after numpy 1.21.0 typedict was depreciated, can you post the full warning? I'm trying to see if another dependency did not update their numpy and maybe that's the error? If not, installing numpy 1.21.0 is a temporary workaround, but before that, if you post the full error.

**Reply by rakeshghosh313** [reply]
Thanks for your comment.
Here I am attaching the full error output.
''
import pyart

## You are using the Python ARM Radar Toolkit (Py-ART), an open source
## library for working with weather radar data. Py-ART is partly
## supported by the U.S. Department of Energy as part of the Atmospheric
## Radiation Measurement (ARM) Climate Research Facility, an Office of
## Science user facility.
##
## If you use this software to prepare a publication, please cite:
##
##     JJ Helmus and SM Collis, JORS 2016, doi: 10.5334/jors.119

Traceback (most recent call last):

  File "<ipython-input-82-413608f118be>", line 1, in <module>
    import pyart

  File "C:\ProgramData\Anaconda3\lib\site-packages\pyart\__init__.py", line 57, in <module>
    from . import aux_io

  File "C:\ProgramData\Anaconda3\lib\site-packages\pyart\aux_io\__init__.py", line 22, in <module>
    from .odim_h5 import read_odim_h5

  File "C:\ProgramData\Anaconda3\lib\site-packages\pyart\aux_io\odim_h5.py", line 10, in <module>
    import h5py

  File "C:\ProgramData\Anaconda3\lib\site-packages\h5py\__init__.py", line 46, in <module>
    from ._conv import register_converters as _register_converters

  File "h5py\h5t.pxd", line 14, in init h5py._conv

  File "h5py\h5t.pyx", line 293, in init h5py.h5t

  File "C:\ProgramData\Anaconda3\lib\site-packages\numpy\__init__.py", line 320, in __getattr__
    raise AttributeError("module {!r} has no attribute "

AttributeError: module 'numpy' has no attribute 'typeDict'
"

**Reply by zssherman** [reply]
@rakeshghosh313 What version is your h5py? Could be outdated. Also are you using a conda environment? Or using the base environment?

**Reply by rakeshghosh313** [reply]
The latest version of h5py 
I am using the base environment.

**Comment by zssherman** [comment]
Closing discussion for now.
